### DISCUSSION TITLE
migrating from amavis to rspamd?

### DISCUSSION THREAD

**Discussion by xinomilo** [modoboa/modoboa#2715] [Help]
to start, is this supported by modoboa? (modoboa-rspamd has no description set...) 
can amavis/spamassassin be completely removed and replaced by rspamd? 
ideally, that would remove opendkim, too, since rspamd can do dkim signing as well..

i've performed such a migration on other systems, but i'm not confident on simply migrating/removing parts in a modoboa system.. 
so do you think this can be done without messing up modoboa+updates? 
if yes, how? (any hints much appreciated... )

thanks,

**Comment by tonioo** [comment]
@xinomilo Yes, you can completely replace amavis/SA/opendkim by rspamd (this is what I did on my own server). And the modoboa plugin for rspamd is there to ease DKIM signing.

**Reply by PatTheMav** [reply]
@tonioo would it be possible to fix the `setup.py` in `modoboa-rspamd` (as it's obviously not compatible with recent changes to `setuptools` anymore)?

I've just switched to rspamd and successfully migrated the DKIM keys, but it would be great if this module could see a slight update for 2023.

EDIT: Seems to only require the changes that were made to modoboa-dmarc's `setup.py` (as long as pip still supports these "legacy" installs). Might open a PR for it after new years'.
